api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java

```diff
@@ -333,17 +334,36 @@ public QueryJobConfiguration previewBigQueryJobConfig(DataSetPreviewRequest requ
 
   @Override
   public Map<String, QueryJobConfiguration> domainToBigQueryConfig(DataSetRequest dataSetRequest) {
+    DbDataset dbDataset;
+    if (CollectionUtils.isEmpty(dataSetRequest.getDomainValuePairs())) {
+      dbDataset =
+          dataSetDao
+              .findDbDatasetByNameAndWorkspaceId(
+                  dataSetRequest.getName(), dataSetRequest.getWorkspaceId())
```

Comment:
We should probably use primary key to do lookups. Ex: dataSetDao.findOne(dataSetId)

Reply:
Done